### PR TITLE
Refactor: no unwrap for arrays

### PR DESCRIPTION
**Pull request by snewcomer**
ref #104 
close #109

### CODE REVIEW

**Comment by snewcomer** [comment]
@NullVoxPopuli

File: src/index.ts [hunk 1]
```diff
@@ -1007,6 +992,25 @@ export class BufferedChangeset implements IChangeset {
...
+      }
+
+      if (isObject(subChanges)) {
+        if (isObject(subContent)) {
+          subChanges = normalizeObject(subChanges, this.isObject);
+          return { ...subContent, ...subChanges };
+        } else if (Array.isArray(subContent)) {
+          subChanges = normalizeObject(subChanges, this.isObject);
+
+          return objectToArray(mergeDeep(arrayToObject(subContent), subChanges));
```
Comment (NullVoxPopuli) [inline src/index.ts:1009]: :+1:

File: src/utils/object-tree-node.ts [hunk 2]
```diff
@@ -130,10 +130,6 @@ class ObjectTreeNode implements ProxyHandler {
 
     return changes;
   }
-
-  toJSON() {
```
Comment (NullVoxPopuli) [inline src/utils/object-tree-node.ts:134]: :+1:

**Comment by NullVoxPopuli** [comment]
looks good! I'm glad this wasn't a lot of code changes

**Review by NullVoxPopuli — APPROVED** [review]

**Pull request merged**